### PR TITLE
feat: Adds SASL (plain and gssapi) support for kafka sink

### PR DESCRIPTION
**Pull request by alexanghh**
Adds SASL (plain and GSSAPI/kerberos) support for Kafka sink.

SCRAM-SHA-256/512 and OAUTH not implemented.
Tested the following SASL mechanism using kafka-docker-playground:

sasl-plain
sasl-ssl
kerberos (keytab and password)
ssl-kerberos (keytab and password)

### CODE REVIEW

**Comment by alexanghh** [comment]
The test / E2E Tests (jetstream, sdks-e2e) failure originates from the main branch.

**Comment by vigith** [comment]
thank you, @alexanghh. could you please add this option to the kafka [source](https://github.com/numaproj/numaflow/blob/main/docs/user-guide/sources/kafka.md) and [sink](https://github.com/numaproj/numaflow/blob/main/docs/user-guide/sinks/kafka.md) documentation too?

File: pkg/apis/numaflow/v1alpha1/kafka_sink.go [hunk 1]
```diff
@@ -25,4 +25,8 @@ type KafkaSink struct {
 	TLS *TLS `json:"tls" protobuf:"bytes,3,opt,name=tls"`
 	// +optional
 	Config string `json:"config,omitempty" protobuf:"bytes,4,opt,name=config"`
+	// SASL user to configure SASL connection for kafka broker
+	// SASL.enable=true default for SASL.
+	// +optional
+	SASL *SASL `json:"sasl" protobuf:"bytes,6,opt,name=sasl"`
```
Comment (whynowy) [inline pkg/apis/numaflow/v1alpha1/kafka_sink.go:31]: Index is incorrect.

Comment (alexanghh) [reply pkg/apis/numaflow/v1alpha1/kafka_sink.go]: Fixed.

**Comment by alexanghh** [comment]
@vigith Added the docs.

**Review by vigith — APPROVED** [review]

**Review by whynowy — APPROVED** [review]

**Pull request merged**